app/src/js/components/Datepicker/Datepicker.js

```diff
@@ -1,103 +1,94 @@
     this.handleHourFormatChange = this.handleHourFormatChange.bind(this);
     this.handleDateTimeRangeChange = this.handleDateTimeRangeChange.bind(this);
     this.clear = this.clear.bind(this);
   }
 
   clear () {
-    this.onChange({...Datepicker.defaultProps});
+    const { value, label } = allDateRanges.find(a => a.label === 'All');
+    this.props.dispatch(this.dispatchDropdownUpdate(value, label));
   }
 
-  handleDateRangeChange (e) {
-    const dateRange = e.target.value;
+  dispatchDropdownUpdate(value, label) {
+    return (dispatch, getState) => {
+      dispatch({
+        type: DATEPICKER_DROPDOWN_FILTER,
+        data: { dateRange: { value, label } }
+      });
+      const datepickerState = getState().datepicker;
+      this.updateRouterWithNewProps(datepickerState);
+    };
+  }
 
-    // set dateRange and clear the dateTimeRange selection
-    this.onChange({...Datepicker.defaultProps, dateRange});
+  handleDropdownChange (e) {
+    const { value, label } = allDateRanges[e.target.selectedIndex];
+    this.props.dispatch(this.dispatchDropdownUpdate(value, label));
   }
 
   handleHourFormatChange (e) {
-    this.onChange({[e.target.name]: e.target.value});
+    this.props.dispatch({
+      type: DATEPICKER_HOUR_FORMAT,
+      data: e.target.value
+    });
   }
 
   handleDateTimeRangeChange (name, newValue) {
-    // The user input is in UTC time zone, but the DateTimePicker takes it as local time.
-    // So we need convert the Date object to UTC time zone by ignoring the time zone offset.
-    const utcValue = moment.utc(moment(newValue).format(dateTimeFormat)).format();
+    // User input is in UTC, but the DateTimePicker component interprets it's
+    // data as local time.  So we need convert the Date value to UTC.
+    const utcValue = moment.utc(moment(newValue).format(dateTimeFormat)).toDate();
+    if (isNaN(utcValue.valueOf())) return;
     const updatedProps = {
       startDateTime: this.props.startDateTime,
       endDateTime: this.props.endDateTime,
       [name]: utcValue
     };
-
-    // only reset the date range if both start and end are selected
-    const { startDateTime, endDateTime } = updatedProps;
-    if (startDateTime && endDateTime) {
-      updatedProps.dateRange = defaultDateRange;
-    }
-    this.onChange(updatedProps);
+    updatedProps.dateRange = allDateRanges.find(a => a.label === 'Custom');
+    this.props.dispatch({type: DATEPICKER_DATECHANGE, data: updatedProps});
+    this.updateRouterWithNewProps(updatedProps);
   }
 
-  onChange (value) {
-    this.props.onChange(value);
+  updateRouterWithNewProps (newProps) {
+    const urlDateFormat = 'YYYYMMDDHHmmSS';
+    const urlProps = ['endDateTime', 'startDateTime'];
+
+    urlProps.map((time) => {
+      let urlValue = '';
+      if (newProps[time] !== null) {
+        urlValue = moment.utc(newProps[time]).format(urlDateFormat);
+      }
+      updateRouterLocation(this.props.router, this.props.location, time, urlValue);
```

Comment:
With the new webpack stuff, `udpateRouterLocation` will no longer work. Instead, we are using the higher order component `withQueryParams` (https://github.com/woltapp/react-router-query-params). We should keep that mind if this gets merged before https://github.com/nasa/cumulus-dashboard/pull/621/

Reply:
Thanks, this is coming after, I'll check on that.